dashboard/app/controllers/ai_diff_controller.rb

```diff
@@ -17,6 +18,45 @@ def chat_completion
 
     response_body = get_response_body(session_id)
 
+    # get or create thread obj
+    begin
+      @thread = session_id ? AichatThread.where(external_id: session_id, user_id: current_user.id).first : AichatThread.create!(
+        user_id: current_user.id,
+        external_id: response_body[:session_id],
+        llm_version: AiDiffBedrockHelper::MODEL_ID,
+        unit_id: @unit.id,
+        level_id: @lesson.id,
+      )
```

Comment:
Would this work? Seems like it might avoid the possible issue where if the request supplies a session ID but the thread doesn't exist for whatever reason, it would create the thread instead of letting `@thread` be nil.
```suggestion
      @thread = AichatThread.find_or_create_by!(
        user_id: current_user.id,
        external_id: response_body[:session_id],
        llm_version: AiDiffBedrockHelper::MODEL_ID,
        unit_id: @unit.id,
        level_id: @lesson.id,
      )
```